### PR TITLE
Adding method to get analytic ID by name + version

### PR DESCRIPTION
**Pull request by brimoor**
Users run analytics via their `name` and `version`, so looking up analytics in this way is also a common use case that we should support.

Note that I am just implementing this clientside for now.

### CODE REVIEW

**Comment by brimoor** [comment]
Depends on how we're defining "API error". There was a circular dependency issue, but the checks that are now `ValueError`s in this PR are happening purely clientside. Should that "count" as an APIError with an appropriately invented "status code"? Or just use builtin error types?

I'm okay with either

**Comment by lethosor** [comment]
The one you changed in query.py corresponds to an API-side error (i.e. if that check+error were removed from query.py entirely, an identical error would be produced from the API). I think it should continue to use APIError for consistency. The same can be said for some existing APIErrors that are thrown client-side, e.g. the ones from 94c3cdca.

Whether the error you added in this PR should be an APIError is another question - a properly-implemented backend route for this should always return either 1 or 0 analytics (maybe a 404 for the latter?), so the error is really due to a limitation of the current implementation. I'd say a ValueError is fine there for now.

**Comment by brimoor** [comment]
Feel free to take this in whatever direction you want. I'm all for one source of truth, so, if you just want to remove the duplicate clientside errors in question altogether, that's fine by me! Or adopt and move APIError. Up to you

**Comment by lethosor** [comment]
Oh, that's a good point - deferring that check to the API would avoid any issues caused by the supported fields list on the client side being wrong. I'd be ok with getting rid of the valid field check here.

**Comment by brimoor** [comment]
@lethosor do what you will (or won't) to this PR so that we can merge it

File: voxel51/users/api.py [hunk 1]
```diff
@@ -185,6 +186,45 @@ def query_analytics(self, analytics_query):
...
 
+    def get_analytic_id(self, name, version=None):
+        '''Gets the ID of the analytic with the given name (and optional
+        version).
+
+        Under the hood, this method uses an
+        :class:`voxel51.users.query.AnalyticsQuery` to search for the analytic
+        on the Platform. Thus, technically, `name` and `version` can be
+        substrings of the full analytic name and version. However, this method
+        raises an error if multiple analytics are found to match the search.
```
Comment (lethosor) [inline voxel51/users/api.py]: This also raises an error if an analytic's name is a substring of another's - e.g. I have `foo` and `foo-bar`, so I can't search for `foo`.

**Review by lethosor — APPROVED** [review]
Works for me now, feel free to adjust

File: voxel51/users/api.py (hunk 1 above)
Comment (brimoor) [reply voxel51/users/api.py]: yeah good point. I better add explicit logic to always allow exact matches

**Pull request merged**